amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/AwsCredentialsProviderPropertyValueDecoder.java

```diff
@@ -79,65 +81,133 @@ private static List<AwsCredentialsProvider> getValidCredentialsProviders(List<St
-                clazz = (Class<? extends AwsCredentialsProvider>) c;
-            } catch (ClassNotFoundException cnfe) {
-                // Providers are a product of prefixed Strings to cover multiple
-                // namespaces (e.g., "Foo" -> { "some.auth.Foo", "kcl.auth.Foo" }).
-                // It's expected that many class names will not resolve.
+            final Class<? extends AwsCredentialsProvider> clazz = getClass(nameAndArgs[0]);
+            if (clazz == null) {
                 continue;
             }
-            log.info("Attempting to construct {}", clazz);
-
             AwsCredentialsProvider provider = null;
             if (nameAndArgs.length > 1) {
                 final String[] varargs = Arrays.copyOfRange(nameAndArgs, 1, nameAndArgs.length);
-
-                // attempt to invoke an explicit N-arg constructor of FooClass(String, String, ...)
-                provider = constructProvider(providerName, () -> {
-                    Class<?>[] argTypes = new Class<?>[nameAndArgs.length - 1];
-                    Arrays.fill(argTypes, String.class);
-                    return clazz.getConstructor(argTypes).newInstance(varargs);
-                });
-
+                provider = tryConstructorWithArgs(providerName, clazz, varargs);
                 if (provider == null) {
-                    // attempt to invoke a public varargs/array constructor of FooClass(String[])
-                    provider = constructProvider(providerName, () -> clazz.getConstructor(String[].class)
-                            .newInstance((Object) varargs));
+                    provider = tryCreateWithArgs(providerName, clazz, varargs);
                 }
             }
-
             if (provider == null) {
-                // regardless of parameters, fallback to invoke a public no-arg constructor
-                provider = constructProvider(providerName, clazz::newInstance);
+                provider = tryConstructorWithNoArgs(providerName, clazz);
             }
-
             if (provider == null) {
-                // if still not found, try empty create() method
-                try {
-                    Method createMethod = clazz.getDeclaredMethod("create");
-                    if (Modifier.isStatic(createMethod.getModifiers())) {
-                        provider = constructProvider(providerName, () -> clazz.cast(createMethod.invoke(null)));
-                    } else {
-                        log.warn("Found non-static create() method in {}", providerName);
-                    }
-                } catch (NoSuchMethodException e) {
-                    // No create() method found for class
-                }
+                provider = tryCreateWithNoArgs(providerName, clazz);
             }
-
             if (provider != null) {
                 credentialsProviders.add(provider);
             }
         }
         return credentialsProviders;
     }
 
+    private static AwsCredentialsProvider tryConstructorWithArgs(
```

Comment:
Can we combine tryConstructorWithArgs with tryConstructorWithNoArgs into one method since they take in similar arguments? 

We can simplify the return logic like this 
```
AwsCredentialsProvider provider = constructProvider(providerName, () -> getConstructorWithVarArgs(clazz, varargs));
if (provider == null) {
    provider = constructProvider(providerName, () -> getConstructorWithArgs(clazz, varargs));
}
if (provider == null) {
    provider = constructProvider(providerName, () -> getCreateMethod(clazz));
}
return provider;
```

We can apply this to the Create related methods too.

Reply:
Addressed, simplifying overall logic significantly.